File: WebView/JsBridgeDemo/app/src/main/java/com/zhewen/jsbridgedemo/MainJavascriptInterface.java
package com.zhewen.jsbridgedemo;

import android.util.Log;
import android.webkit.JavascriptInterface;

import androidx.annotation.NonNull;

import com.zhewen.jsbridge.BaseJavascriptInterface;
import com.zhewen.jsbridge.BridgeHelper;
import com.zhewen.jsbridge.BridgeWebView;

/**
 * Created on 2019/7/10.
 * Author: bigwang
 * Description:
 */
public class MainJavascriptInterface extends BaseJavascriptInterface {


    public MainJavascriptInterface(@NonNull BridgeWebView webView) {
        super(webView);
    }

    public MainJavascriptInterface(BridgeHelper helper) {
        super(helper);
    }


    public String send(String data) {
        return "it is default response";
    }


    @JavascriptInterface
    public String submitFromWeb(String data, String callbackId) {
        Log.d("chromium data", data + ", callbackId: " + callbackId + " " + Thread.currentThread().getName());
//        mWebView.sendResponse("submitFromWeb response", callbackId);
        return "submitFromWeb response";
    }
}

File: OpenSourceLibrary/JsBridgeLab/app/src/main/java/com/zhewen/jsbridgelab/MainJavascriptInterface.kt
package com.zhewen.jsbridgelab

import android.util.Log
import android.webkit.JavascriptInterface
import android.widget.Toast
import com.zhewen.jsbridge.jsinterface.BaseJavascriptInterface
import com.zhewen.jsbridge.jsinterface.OnBridgeCallback
import com.zhewen.jsbridge.webview.BridgeWebView

open class MainJavascriptInterface(callbacks:HashMap<String,OnBridgeCallback>, webView:BridgeWebView):BaseJavascriptInterface(callbacks) {

    private val mWebView:BridgeWebView = webView

    override fun toast(data: String) {
        application?.let {
            Toast.makeText(it,"data from js %data",Toast.LENGTH_LONG).show()
        }
    }

    override fun phoneInfo(data: String?) {
        Log.d("MainJavascriptInterface","data = $data")
    }

    @JavascriptInterface
    open fun submitFromWeb(data: String, callbackId: String) {
        Log.d("chromium data", data + ", callbackId: " + callbackId + " " + Thread.currentThread().name)
        mWebView.sendResponse("submitFromWeb response", callbackId)
    }

}